### ISSUE TITLE
VariableName cop raises "undefined method `loc'"

### ISSUE THREAD

**Issue by roberts1000** [rubocop-hq/rubocop-rspec#957]
The `VariableName` cop is raising this error:

```
An error occurred while RSpec/VariableName cop was inspecting /home/dev/Projects/my_project/spec/libs/foo/foo_spec.rb:68:4.
undefined method `loc' for #<Parser::Source::Range:0x0000000003abdbe8>
/home/dev/.rvm/gems/ruby-2.7.1/gems/rubocop-0.87.1/lib/rubocop/cop/cop.rb:41:in `find_location'
/home/dev/.rvm/gems/ruby-2.7.1/gems/rubocop-0.87.1/lib/rubocop/cop/cop.rb:29:in `add_offense'
/home/dev/.rvm/gems/ruby-2.7.1/gems/rubocop-0.87.1/lib/rubocop/cop/mixin/configurable_formatting.rb:13:in `check_name'
```

This code triggers the error:

```ruby
let(:one_OC) { [] }
````

This works:

```ruby
let(:one_oc) { [] }
````

```
$ rubocop -V
0.87.1 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.1 x86_64-linux)
```

```
$ gem list rubocop-performance

*** LOCAL GEMS ***

rubocop-performance (1.7.0)
```

**Comment by reidab** [comment]
Same issue here; this is also triggered by:

```ruby
let(:'canonical-record-id') { 123 }
```

**Comment by bquorning** [comment]
I haven’t attempted to reproduce the issue, but it looks like it may be caused by some Changes to RuboCop’s internal API in version 0.87.0. We’re looking to fix this incompatibility in #955

**Comment by jtannas** [comment]
+1; being triggered by
```ruby
let(:'id[]') { ['foo', 'bar', 'baz'] }
```

P.S. The weird variable name is explained in Issue #941. I'll be working on a PR for the feature once this issue is resolved.
